platform/x11/cog-platform-x11.c

```diff
@@ -643,16 +591,16 @@ init_egl (void)
         return FALSE;
 
     Window win = (Window) s_window->xcb.window;
-    s_display->egl.create_platform_window_surface = load_egl_proc_address ("eglCreatePlatformWindowSurfaceEXT");
-    if (s_display->egl.create_platform_window_surface)
-        s_window->egl.surface = s_display->egl.create_platform_window_surface (s_display->egl.display,
-                                                                               s_display->egl.config,
-                                                                               &win, NULL);
+    if (epoxy_has_egl_extension(s_display->egl.display, "EGL_EXT_platform_x11")) {
+        s_window->egl.surface =
+            eglCreatePlatformWindowSurfaceEXT(s_display->egl.display, s_display->egl.config, &win, NULL);
+    } else {
+        s_window->egl.surface =
+            eglCreateWindowSurface(s_display->egl.display, s_display->egl.config, (EGLNativeWindowType) &win, NULL);
+    }
```

Comment:
I guess this is if we get given a display without the X11 extension, we still end up with a surface... But do we just not need the X11 extension? We don't look at `create_platform_window_surface` anymore, this seems like a change in behaviour and I don't really understand it - maybe a comment here is warranted?

Reply:
The fallback is intended to be used for drivers which do not provide the extension (typically old-ish ones), assuming that `eglGetDisplay()` would have returned a correct kind of display (an X11 display in this case), then `eglCreateWindowSurface()` will work just fine... but it is now that I notice that we never do in the code a fallback from `eglGetPlatformDisplayEXT()` to `eglGetDisplay()` if the `EGL_EXT_platform_x11` extension is not available, so the fallback here is dead code and will never be reached. Let's remove it.

Reply:
Also: we may want to `g_error()` inside `init_egl()` if `eglGetPlatformDisplayEXT()` cannot be used instead of continuing without a valid `EGLDisplay`. I'll add this.